src/analysis/utils/disk-utils.ts

```diff
@@ -44,7 +44,9 @@ export const getCurrentAttachedDataDisk = (
   app: App,
   appDataDisks: PersistentDisk[]
 ): DecoratedPersistentDisk | undefined => {
-  const currentDisk: PersistentDisk | undefined = _.find({ name: app?.diskName }, appDataDisks);
```

Comment:
Can the diskName be null because WDS does not use persistent disk?

Reply:
I'm not sure. The current types have the `diskName` field typed as potentially absent and if it is present, string or undefined. Since the dev API returned `null` for `diskName` for a WDS app, I'm wondering if that field is always present and would be accurately typed as string or null?

Reply:
The typing of `diskName` reflects the backend fact that the diskName is `Option[String]` in scala. I believe our current scala decoder puts null in this field instead of omitting it, but I'd rather not put that implementation detail in the UI

Reply:
I don't think that's an implementation detail. The UI only sees the serialized value, so the behavior of the decoder affects the API contract between Leo and the UI.

TS can verify how data flows and is transformed through the UI, but a lot of the safety that provides is based on how accurately we represent the types of inputs. If our API response types aren't accurate, then it calls into question the type safety of all the code that uses those responses.

For example,
```ts
const app: ListAppResponse = ...;
const diskName = app.diskName === undefined ? undefined : app.diskName.toLowerCase();
```

With the current `ListAppResponse` type, TS is happy with this. But if `app.diskName` may be null, there's a bug here since we may be calling `.toLowerCase()` on null.

If we want to have TS types for API responses mirror the types of backend models, I'd suggest we use our own utility type.

```ts
type Optional<T> = T | null;

export interface ListAppResponse {
  ...
  diskName: Optional<string>;
  ...
}
```

@msilva-broad thoughts?

Reply:
agreed that the types need to reflect reality... Scala Option[] mechanics or whatever don't matter as much as the end result being whatever JSON shape is generated.  If it's string | null, then that is the hard contract and should be reflected as such in the Typescript definition.

Reply:
Ahh... ok, this is a larger problem then. I'll make a ticket on the IA board to handle this since it affects most of our models. cc @LizBaldo , this is why there is a `diskName` `null` check. We may want to investigate having the decoder omit the field rather than returning null... though a UI change is probably more appropriate...

Reply:
Ok. I think I'll remove the app model changes from this PR, add a `ts-expect-error` where necessary.

Reply:
Filed [IA-4439](https://broadworkbench.atlassian.net/browse/IA-4439) for this.